accept UTF8-BOM when reading dashboard or monitor from JSON file

kmuto opened the pull request:
Passing a JSON file with a UTF-8 BOM to `mkr dashboards push --file-path` causes an error that is difficult for users to understand.

```
     error invalid character 'ï' looking for beginning of value
```

I understand that "JSON files must not contain a BOM", but I thought it would be more user-friendly to ignore the BOM part and read the file without making an error, so I made a patch.

Test
```
# original version 0.54.0
$ mkr dashboards push --file-path t-nobom.json
$ mkr dashboards push --file-path t-withbom.json
     error invalid character 'ï' looking for beginning of value

# (remove dashboards)

# patched version
$ ./mkr dashboards push --file-path t-nobom.json
$ ./mkr dashboards push --file-path t-withbom.json
```
[t-nobom.json](https://github.com/mackerelio/mkr/files/13956859/t-nobom.json)
[t-withbom.json](https://github.com/mackerelio/mkr/files/13956861/t-withbom.json)

lufia commented on dashboards/command.go:
```diff
@@ -92,8 +95,10 @@ func doPushDashboard(c *cli.Context) error {
 	f := c.String("file-path")
 	src, err := os.Open(f)
 	logger.DieIf(err)
+	fallback := unicode.UTF8.NewDecoder()
+	r := transform.NewReader(src, unicode.BOMOverride(fallback))
```
I do like it a lot!

lufia reviewed (commented):
Could you add a test?

kmuto commented:
I added similar BOM handling to monitors code. (it will be called by `mkr monitors diff` or `mkr monitors push`)
I added the test unit to monitors/command_test.go also, but gave up to create a test for dashboards because `doPushDashboard()` is too large to only test BOM.

lufia commented on monitors/command_test.go:
```diff
@@ -200,3 +200,36 @@ func TestDiffMonitorsWithScopes(t *testing.T) {
 	}
 }
+
+func TestMonitorLoadRules(t *testing.T) {
```
for example,

```suggestion
func TestMonitorLoadRulesWithBOM(t *testing.T) {
```

because this function tests if it ignore BOM.

lufia commented on monitors/command_test.go:
```diff
@@ -200,3 +200,36 @@ func TestDiffMonitorsWithScopes(t *testing.T) {
 }
+
+func TestMonitorLoadRules(t *testing.T) {
+	tmpFile, err := os.CreateTemp("", "")
+	if err != nil {
+		t.Errorf("should not raise error: %v", err)
+	}
+	defer os.Remove(tmpFile.Name())
```
testing.T.TempDir is useful.

kmuto replied:
I had tried it on https://github.com/mackerelio/mkr/pull/622/commits/8df6538fbd93beddeec7de718d1282d5f200910d , https://github.com/mackerelio/mkr/pull/622/commits/3295eed0f77b3460d2fe25143831d47360b02eae , and https://github.com/mackerelio/mkr/pull/622/commits/db949f894855a61decdeccff129df617a33e9012 but all of them went failed on Windows (I saw https://github.com/golang/go/issues/50510 )
It would be better to use it but I keep CreateTemp code at this time for Windows.
